src/core/Akka/Actor/Futures.cs

```diff
@@ -26,17 +26,37 @@ public static class Futures
         //when asking from outside of an actor, we need to pass a system, so the FutureActor can register itself there and be resolvable for local and remote calls
         public static Task<object> Ask(this ICanTell self, object message, TimeSpan? timeout = null)
         {
-            return self.Ask<object>(message, timeout);
+            return self.Ask<object>(message, timeout, CancellationToken.None);
+        }
+
+        public static Task<object> Ask(this ICanTell self, object message, CancellationToken cancellationToken)
+        {
+            return self.Ask<object>(message, null, cancellationToken);
+        }
+
+        public static Task<object> Ask(this ICanTell self, object message, TimeSpan? timeout, CancellationToken cancellationToken)
+        {
+            return self.Ask<object>(message, timeout, cancellationToken);
         }
 
         public static Task<T> Ask<T>(this ICanTell self, object message, TimeSpan? timeout = null)
         {
+            return self.Ask<T>(message, timeout, CancellationToken.None);
+        }
+
+        public static Task<T> Ask<T>(this ICanTell self, object message, CancellationToken cancellationToken)
+        {
+            return self.Ask<T>(message, null, cancellationToken);
+        }
+
+        public static Task<T> Ask<T>(this ICanTell self, object message, TimeSpan? timeout, CancellationToken cancellationToken)
```

Comment:
timeout + cancellationToken is redundant.

If the caller really needs to combine a timeout and a CancellationToken, he can use [CancellationTokenSource.CreateLinkedTokenSource](https://msdn.microsoft.com/en-us/library/dd642265(v=vs.110).aspx)